snuba/datasets/schemas/join_schema.py

```diff
@@ -0,0 +1,90 @@
+from dataclasses import dataclass
+from enum import Enum
+from typing import Callable, Mapping, Optional, Sequence
+
+from snuba.clickhouse.columns import ColumnSet
+from snuba.datasets.schemas import Schema
+
+
+class JoinType(Enum):
+    INNER = "INNER"
+    LEFT = "LEFT"
+    RIGHT = "RIGHT"
+    FULL = "FULL"
+
+
+@dataclass(frozen=True)
+class JoinMapping():
+    left_alias: str
```

Comment:
This name is a little confusing, it's ambiguous whether or not it's a table alias or column alias without reading the implementation.